vulture/config.py

```diff
@@ -0,0 +1,212 @@
+        f' "*settings.py,docs/*.py"). {glob_help} A PATTERN without glob'
+        f" wildcards is treated as *PATTERN*.",
+    )
+    parser.add_argument(
+        "--ignore-decorators",
+        metavar="PATTERNS",
+        type=csv,
+        default=MISSING,
+        help=f"Comma-separated list of decorators. Functions and classes using"
+        f' these decorators are ignored (e.g., "@app.route,@require_*").'
+        f" {glob_help}",
+    )
+    parser.add_argument(
+        "--ignore-names",
+        metavar="PATTERNS",
+        type=csv,
+        default=MISSING,
+        help=f'Comma-separated list of names to ignore (e.g., "visit_*,do_*").'
+        f" {glob_help}",
+    )
+    parser.add_argument(
+        "--make-whitelist",
+        action="store_true",
+        default=MISSING,
+        help="Report unused code in a format that can be added to a"
+        " whitelist module.",
+    )
+    parser.add_argument(
+        "--min-confidence",
+        type=int,
+        default=MISSING,
+        help="Minimum confidence (between 0 and 100) for code to be"
+        " reported as unused.",
+    )
+    parser.add_argument(
+        "--sort-by-size",
+        action="store_true",
+        default=MISSING,
+        help="Sort unused functions and classes by their lines of code.",
+    )
+    parser.add_argument(
+        "-v", "--verbose", action="store_true", default=MISSING
+    )
+    parser.add_argument("--version", action="version", version=version)
+    namespace = parser.parse_args(args)
+    return from_dict(vars(namespace))
+
+
+def make_config(argv=None, tomlfile=None):
+    """
+    Returns a config object for vulture, merging both ``pyproject.toml`` and
+    CLI arguments (CLI will have precedence).
+
+    :param argv: The CLI arguments to be parsed. This value is transparently
+        passed through to :py:meth:`argparse.ArgumentParser.parse_args`
+    :param tomlfile: An IO instance containing TOML data. By default this will
+        auto-detect an existing ``pyproject.toml`` file and exists solely for
+        unit-testing.
+    """
+    # If we loaded data from a TOML file, we want to print this out on stdout
+    # in verbose mode so we need to keep the value around.
+    detected_toml_path = ""
+
+    if tomlfile:
+        config = _parse_toml(tomlfile)
+        detected_toml_path = str(tomlfile)
+    else:
+        toml_path = abspath("pyproject.toml")
+        if exists(toml_path):
+            with open(toml_path) as config:
+                config = _parse_toml(config)
+            detected_toml_path = toml_path
+        else:
+            config = {}
+
+    # We can't use a simple call to dict.update() because some values should
+    # not be overwritten in the config. More precisely, if the values have "no
+    # real default" in the CLI args, they should not be taken into
+    # consideration.
+    cli_config = _parse_args(argv)
+    for key in config.keys():
```

Comment:
I think it's clearer to use `for key, value in cli_config.items():` here.

Reply:
That won't work, because I don't *read* `value` from the dict. Instead I am using `key` to assign the CLI-args value into the dict.

Reply:
I still think that the following code is identical and a bit simpler:

```
for key, value in cli_config.items():
    if value is not MISSING:
        config[key] = value
```

Am I missing something?

Reply:
You're right. I misread your initial code snippet (mistook `cli_config.items` for `config.items()`